### PR TITLE
CCR Careers automation skeleton and login method

### PR DESCRIPTION
**Pull request by alenaminsk**

### CODE REVIEW

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package CustomBusiness.BusinessAppRunners;
+
+import java.io.File;
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java:4]: Please remove the empty line

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package CustomBusiness.BusinessAppRunners;
+
+
+import java.io.File;
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java:5]: Please remove the empty line#5

File: src/test/java/CustomBusiness/CCR/Features/LandingPage.feature [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Feature: CCR Landing Page Scenarios
+
+  Background: common steps
+    Given User is on CCR Landing page and user is "internal user"
+
+  @Alena @Regression @Smoke @Progression @landingPage
```
Comment (zamant2) [inline src/test/java/CustomBusiness/CCR/Features/LandingPage.feature:6]: @Progression tag shouldn't merge to master

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package CustomBusiness.BusinessAppRunners;
+
+
+import java.io.File;
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java:5]: Please remove the empty line#5

File: src/test/java/CustomBusiness/CCR/Pages/CCRLandingPage.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    /** Home Tab */
+    @FindBy(xpath = "//ul[@class='nav navbar-nav']//a[contains(text(),'Home')]")
+    public WebElement homeTab;
+
+    /** Positions Tab */
+    @FindBy(xpath = "//ul[@class='nav navbar-nav']//a[contains(text(),'Positions')]")
+    public WebElement positionsTab;
+
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/CCR/Pages/CCRLandingPage.java:23]: Remove all empty lines please

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+import com.nci.automation.web.ConfUtils;
+import com.nci.automation.web.WebDriverUtils;
+import com.nci.automation.xceptions.TestingException;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.Scenario;
+import org.apache.commons.lang.StringUtils;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java:19]: Please Remove empty line#19

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    @After
+    public void genericTearDown(Scenario s) throws TestingException {
+        if (s.isFailed()) {
+            final byte[] screenshot = ((TakesScreenshot) WebDriverUtils.webDriver).getScreenshotAs(OutputType.BYTES);
+            s.attach(screenshot, "image/png", s.getName());
+        }
+        if (WebDriverUtils.webDriver != null) {
+            MiscUtils.sleep(2000);
+
```
Comment (zamant2) [inline src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java:82]: Please remove empty line#82

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java (hunk 1 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java]: removed

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java (hunk 2 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java]: removed

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java (hunk 4 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java]: removed

File: src/test/java/CustomBusiness/CCR/Features/LandingPage.feature (hunk 3 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/Features/LandingPage.feature:6]: removed

File: src/test/java/CustomBusiness/CCR/Pages/CCRLandingPage.java (hunk 5 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/Pages/CCRLandingPage.java:23]: removed

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java (hunk 6 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java:19]: removed

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java (hunk 7 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java:82]: removed

**Review by andreyeve — APPROVED** [review]

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+ * This class is annotated with @RunWith(Cucumber.class) and it will run
+ * cucumber feature(s).
+ *
+ * @author juarezds
+ */
+
+public class RunCCRProgressionTest {
+
+    @BeforeClass
+    public static void runSetup() {
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java:33]: Do we need runSetup() method in this runner class? I don't see it in many of the other classes.

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ * This class is annotated with @RunWith(Cucumber.class) and it will run
+ * cucumber feature(s).
+ *
+ * @author juarezds
+ */
+
+public class RunCCRRegressionTest {
+
+    @BeforeClass
+    public static void runSetup() {
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java:33]: Do we need runSetup() method in this runner class? I don't see it in many of the other classes.

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ * This class is annotated with @RunWith(Cucumber.class) and it will run
+ * cucumber feature(s).
+ *
+ * @author juarezds
+ */
+
+public class RunCCRSmokeTest {
+
+    @BeforeClass
+    public static void runSetup() {
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java:33]: Do we need runSetup() method in this runner class? I don't see it in many of the other classes.

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+        // use this for web specific setup
+        System.out.println("web specific setup");
+    }
+
+    @After("@web")
+    public void webTearDown() throws MalformedURLException {
+        // use this for web specific clean up
+        System.out.println("web specific clean up");
+    }
+
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java]: Do we need the empty line (112)?

File: src/test/java/CustomBusiness/CCR/StepsImplementation/CCRLogInStepsImplementation.java [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+public class CCRLogInStepsImplementation extends PageInitializer {
+    public void ccrLogin() throws TestingException {
+        nativeViewLoginImpl.sideDoorAccountLogin();
+        MiscUtils.sleep(2000);
+        WebDriverUtils.webDriver.get(EnvUtils.getApplicationUrl("CCR"));
+        MiscUtils.sleep(2000);
+    }
+}
+
```
Comment (hamidnesar) [inline src/test/java/CustomBusiness/CCR/StepsImplementation/CCRLogInStepsImplementation.java:17]: Empty line

**Review by georgeb79 — APPROVED** [review]

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java (hunk 8 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRProgressionTest.java]: Removed runSetup() method

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java (hunk 9 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRRegressionTest.java]: runSetup() method

File: src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java (hunk 10 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/BusinessAppRunners/RunCCRSmokeTest.java]: runSetup() method

File: src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java (hunk 11 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/Steps/HooksSteps.java]: removed

File: src/test/java/CustomBusiness/CCR/StepsImplementation/CCRLogInStepsImplementation.java (hunk 12 above)
Comment (alenaminsk) [reply src/test/java/CustomBusiness/CCR/StepsImplementation/CCRLogInStepsImplementation.java]: removed

**Review by hamidnesar — APPROVED** [review]

**Review by zamant2 — APPROVED** [review]

**Review by hamidnesar — APPROVED** [review]

**Pull request merged**